include/envoy/singleton/manager.h

```diff
@@ -18,6 +18,13 @@ class Registration {
 public:
   virtual ~Registration() = default;
   virtual std::string name() PURE;
+
+  /**
+   * @return std::string the identifying category name for objects
+   * created by this factory. Used for automatic registration with
+   * FactoryCategoryRegistry.
+   */
+  static std::string category() { return "singletons"; }
```

Comment:
This is a bit different and not really part of the main registration system. Does it fail without this change? That would surprise me? If it does I wonder if we should hide this somehow as I don't think it will be relevant to end users.

Reply:
Right. I didn't know about the singleton factory at the time, but this was the kind of thing I had in mind when I originally made the registration opt-in. I can imagine that presence of singletons could indicate the presence of extensions that are not otherwise detectable, so that is a benefit, but of course you could equally argue that it exposes too much of the internals.

With the current patch, anything that registers a factory is required to have a category and is exposed in the logs. Since the singleton manager used a factory, it gets caught up at build time.